crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -2399,18 +2397,9 @@ impl traits::MutTx for Locking {
             .with_label_values(workload, db, reducer)
             .observe(elapsed_time);
 
-        fn hash(a: &WorkloadType, b: &Address, c: &str) -> u64 {
-            use std::hash::Hash;
-            let mut hasher = DefaultHasher::new();
-            a.hash(&mut hasher);
-            b.hash(&mut hasher);
-            c.hash(&mut hasher);
-            hasher.finish()
-        }
-
         let mut guard = MAX_TX_CPU_TIME.lock().unwrap();
         let max_cpu_time = *guard
-            .entry(hash(workload, db, reducer))
+            .entry((*db, *workload, reducer.to_owned()))
```

Comment:
This was an optimization to avoid extra allocations when recording metrics. But I'd rather take the allocations than deal with a potential hash collision when debugging a performance issue.